### ISSUE TITLE
Build fails, when processing large images

### ISSUE THREAD

**Issue by AndreasFaust** [gatsbyjs/gatsby#13442]
## Description
Hello Gatsby-Team!
I am working on a designer's portfolio website with lots of large images. These images are downloaded from a server and processed by Gatsby Image. In dev-mode everything works as expected, but when I try to trigger a build, I get messages like this:

```javascript
error UNHANDLED REJECTION

Error: Failed to process image /Users/andreasfaust/websites/lisa/frontend/.cache/gatsby-source-filesystem/0cfb27ee68f5a1304c081d57405030ef/fritzlambada-06.jpg

(sharp:1306): GLib-CRITICAL **: 11:25:32.087: g_hash_table_lookup: assertion 'hash_table != NULL' failed

(sharp:1306): GLib-CRITICAL **: 11:25:32.087: g_hash_table_lookup: assertion 'hash_table != NULL' failed

(sharp:1306): GLib-CRITICAL **: 11:25:32.088: g_hash_table_lookup: assertion 'hash_table != NULL' failed

(sharp:1306): GLib-CRITICAL **: 11:25:32.088: g_hash_table_insert_internal: assertion 'hash_table != NULL' failed

(sharp:1306): GLib-CRITICAL **: 11:25:32.088: g_hash_table_lookup: assertion 'hash_table != NULL' failed
./build.sh: line 9:  1306 Segmentation fault: 11  gatsby build

```

When I reduce the number of images-collections, it starts working, but not necessarily. Image collections seem to work and fail randomly. The images itself are saved-for-web in Photoshop.

To source those images I am using a selfwritten source-plugin called "gatsby-source-custom-api", which is based on "gatsby-source-filesystem".

### Expected result

Images get downloaded and processed and a build is completed.
Maybe the build time is about a couple minutes, but that does not matter.  

### Actual result

The build fails regularly, eventually works, when I disable certain image collections. The configuration to make it work changes. 

### Environment

  System:
    OS: macOS 10.14.3
    CPU: (8) x64 Intel(R) Core(TM) i7-3720QM CPU @ 2.60GHz
    Shell: 5.5.1 - /usr/local/bin/zsh
  Binaries:
    Node: 11.6.0 - /usr/local/bin/node
    Yarn: 1.15.2 - ~/.yarn/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.15 - /usr/local/bin/python
  Browsers:
    Chrome: 73.0.3683.103
    Firefox: 66.0.2
    Safari: 12.0.3
  npmGlobalPackages:
    gatsby-cli: 2.5.4

package.json:
"gatsby": "^2.3.22",
"gatsby-cli": "^2.5.7",
"gatsby-image": "^2.0.38",
"gatsby-plugin-sharp": "^2.0.34",
"gatsby-source-filesystem": "^2.0.27",

**Comment by AndreasFaust** [comment]
After disabling `sharp.simd(true);` like described at [https://github.com/gatsbyjs/gatsby/issues/6291](https://github.com/gatsbyjs/gatsby/issues/6291), this error shows up: 

```bash
Generating image thumbnails [==============================] 6/6 0.6 secs 100%
error UNHANDLED REJECTION


Error: Failed to process image /Users/andreasfaust/websites/lisa/frontend/.cache/gatsby-source-filesystem/68cf86b87a961a36b4d15640c  6e634dd/bm-zeitung-00-showcase.jpg

libc++abi.dylib: terminating with uncaught exception of type std::__1::system_error: mutex lock failed: Invalid argument
./build.sh: line 9:  3392 Abort trap: 6           gatsby build
error Command failed with exit code 134.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by DSchau** [comment]
Could you provide a [reproduction](https://gatsby.dev/reproduction)?

Thank you!

**Comment by divyamrastogi** [comment]
Happened to me too. For some weird reason, I just ran yarn once and it started working fine.

**Comment by AndreasFaust** [comment]
@DSchau Thank you for the response! I followed your guidelines and made a repo to reproduce the errors: https://github.com/AndreasFaust/gatsby-bug

Trying this out I experienced different results: First it completely worked. Then I deleted the` .cache`- and the `public`-folders, triggered build and the build stopped without exiting after the downloading process. The third time I started the build again and got this error:

```bash
error
The GraphQL query from /Users/andreasfaust/websites/bug-repro/src/pages/archive.js failed.

Errors:
  VipsJpeg: Premature end of JPEG file
  VipsJpeg: out of order read at line 48


  GraphQL request (16:13)
  15:           childImageSharp {
  16:             fluid(maxWidth: 1200) {
                  ^
  17:               ...GatsbyImageSharpFluid_withWebp_noBase64
  ,Cannot read property 'format' of null

  GraphQL request (16:13)
  15:           childImageSharp {
  16:             fluid(maxWidth: 1200) {
                  ^
  17:               ...GatsbyImageSharpFluid_withWebp_noBase64

URL path:
  /archive/
Context:
  {}
Plugin:
  none
Query:
  query usersAndreasfaustWebsitesBugReproSrcPagesArchiveJs3345153068 {
    allArchive {
      nodes {
        url
        title
        text
        description
        contentWork {
          type
          videoDesktop
          videoMobile
          url
          alttext
          local {
            childImageSharp {
              fluid(maxWidth: 1200) {
                ...GatsbyImageSharpFluid_withWebp_noBase64
              }
            }
          }
        }
      }
    }
  }

  fragment GatsbyImageSharpFluid_withWebp_noBase64 on ImageSharpFluid {
    aspectRatio
    src
    srcSet
    srcWebp
    srcSetWebp
    sizes
  }
```

**Comment by bchehraz** [comment]
Today I tried updating my Gatsby project to a more up-to-date version (from 2.0.53 to 2.3.31). Along with that, I updated all of its dependencies to their recommended versions. I am now having the same issue. Luckily I can revert back with this project.

I also created a fresh Gatsby JS project (the default starter) to see if it's just my old project or if there's really a problem here. All I did was create the new project, moved the image into the images folder, and replaced the default graphql query to query the correct file name. The same error occurs.

Here's my output from `gatsby info`. I'm also using npm as my package manager, not yarn.

```
  System:
    OS: macOS 10.14.3
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 8.11.3 - ~/.nvm/versions/node/v8.11.3/bin/node
    Yarn: 1.9.4 - /usr/local/bin/yarn
    npm: 5.6.0 - ~/.nvm/versions/node/v8.11.3/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 73.0.3683.103
    Safari: 12.0.3
  npmPackages:
    gatsby: ^2.3.31 => 2.3.31 
    gatsby-image: ^2.0.40 => 2.0.40 
    gatsby-plugin-manifest: ^2.0.29 => 2.0.29 
    gatsby-plugin-offline: ^2.0.25 => 2.0.25 
    gatsby-plugin-react-helmet: ^3.0.12 => 3.0.12 
    gatsby-plugin-sharp: ^2.0.35 => 2.0.35 
    gatsby-source-filesystem: ^2.0.32 => 2.0.32 
    gatsby-transformer-sharp: ^2.1.18 => 2.1.18 
  npmGlobalPackages:
    gatsby-cli: 2.5.12
```

**Comment by alexanderwe** [comment]
I can confirm I have the exact some error with 99% the same use case. I am working on a photography portfolio with images saved by Photoshop's save-for-web feature.  The image which is causing the build to fail changes randomly every time I want to build the site. For me it seems it is always a `.jpg`

I get the same error messages:

```shell
error UNHANDLED REJECTION

Error: Failed to process image /Users/user/Documents/Programming/gatsby-projects/theme/si  te/.cache/gatsby-source-filesystem/62461a03baef7bca3600a0cbdcf7de5d/L1009643.jpg

(sharp:67220): GLib-CRITICAL **: 14:34:59.191: g_hash_table_lookup: assertion 'hash_table != NULL' failed

(sharp:67220): GLib-CRITICAL **: 14:34:59.192: g_hash_table_lookup: assertion 'hash_table != NULL' failed

(sharp:67220): GLib-CRITICAL **: 14:34:59.192: g_hash_table_insert_internal: assertion 'hash_table != NULL' failed

(sharp:67220): GLib-CRITICAL **: 14:34:59.192: g_hash_table_lookup: assertion 'hash_table != NULL' failed
error Command failed with signal "SIGSEGV".
```

With following dependencies:
```
"babel-plugin-styled-components": "1.10.0",
"gatsby-image": "2.0.40",
"gatsby-plugin-manifest": "2.0.29",
"gatsby-plugin-page-creator": "2.0.12",
"gatsby-plugin-react-helmet": "3.0.12",
"gatsby-plugin-sharp": "2.0.35",
"gatsby-plugin-sitemap": "2.0.12",
"gatsby-plugin-styled-components": "3.0.7",
"gatsby-plugin-typography": "2.2.13",
"gatsby-source-instagram": "0.5.0",
"gatsby-source-wordpress": "3.0.57",
"gatsby-transformer-sharp": "2.1.18",
"gatsby-wordpress-inline-images": "1.0.0",
"react-helmet": "5.2.0",
"react-images": "0.5.19",
"react-media": "1.9.2",
"react-photo-gallery": "7.0.2",
"react-typography": "0.16.19",
"styled-components": "4.2.0",
"typography": "0.16.19"
```

And the following output of `Gatsby info`:

```
  System:
    OS: macOS 10.14.4
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 10.15.3 - /var/folders/8g/thhx4y056s92_s9xqfytf6ww0000gn/T/yarn--1556539822116-0.773076380910849/node
    Yarn: 1.15.2 - /var/folders/8g/thhx4y056s92_s9xqfytf6ww0000gn/T/yarn--1556539822116-0.773076380910849/yarn
    npm: 6.4.1 - ~/.nvm/versions/node/v10.15.3/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Firefox: 66.0.3
    Safari: 12.1
```

**Comment by mohoff** [comment]
I'm experiencing the exact same issue. It errors with images of 5000x3000 but succeeds with smaller photos of 1000x600. The used images were stock ones from https://picsum.photos

**Comment by i-bsd** [comment]
Anyone solve this issue yet?

Having the same UNHANDLED REJECTION error on images on FreeBSD. Seems to be any starter that uses gatsby-image.

**Comment by jacobsilver2** [comment]
I'm having similar issues, but not necessarily with large images.  For me the issue seems to be the amount of images (I have around 200 on the site I'm working on).  During the build stage I get "failed to process remote content" for about 20-30 images, which ultimately results in a build fail.  I have yet to find a solution.

**Comment by dante-blitz** [comment]
I'm getting the following error when trying to builda number of gatsby starters

```console
error UNHANDLED REJECTION


  Error: Failed to process image .../content/images/gallery/page-transitions.jpg


(sharp:4136): GLib-CRITICAL **: 15:19:53.070: g_hash_table_lookup: assertion 'hash_table != NULL' failed

(sharp:4136): GLib-CRITICAL **: 15:19:53.285: g_hash_table_lookup: assertion 'hash_table != NULL' failed```

anyone know how to resolve.

gatsby info: 

```console
  System:
    OS: Linux 4.4 Ubuntu 18.04.2 LTS (Bionic Beaver)
    CPU: (8) x64        Intel(R) Xeon(R) CPU E5-1620 0 @ 3.60GHz
    Shell: 4.4.19 - /bin/bash
  Binaries:
    Node: 10.11.0 - ~/.nvm/versions/node/v10.11.0/bin/node
    Yarn: 1.15.2 - ~/.nvm/versions/node/v10.11.0/bin/yarn
    npm: 6.9.0 - ~/.nvm/versions/node/v10.11.0/bin/npm
  Languages:
    Python: 2.7.15 - /usr/bin/python
  npmPackages:
    gatsby: ^2.3.26 => 2.4.3
    gatsby-image: ^2.0.39 => 2.0.41
    gatsby-plugin-eslint: ^2.0.3 => 2.0.5
    gatsby-plugin-offline: ^2.0.25 => 2.1.0
    gatsby-plugin-react-helmet: ^3.0.12 => 3.0.12
    gatsby-plugin-react-svg: ^2.0.0 => 2.1.1
    gatsby-plugin-sharp: ^2.0.35 => 2.0.37
    gatsby-plugin-sitemap: ^2.0.12 => 2.1.0
    gatsby-plugin-styled-components: ^3.0.7 => 3.0.7
    gatsby-plugin-webpack-size: ^0.0.3 => 0.0.3
    gatsby-source-filesystem: ^2.0.30 => 2.0.33
    gatsby-transformer-json: ^2.1.11 => 2.1.11
    gatsby-transformer-remark: ^2.3.9 => 2.3.12
    gatsby-transformer-sharp: ^2.1.18 => 2.1.19
  npmGlobalPackages:
    gatsby-cli: 2.5.8
```

**Comment by i-bsd** [comment]
Me too. On FreeBSD:

```
error UNHANDLED REJECTION


  Error: Failed to process image /usr/home/user/Work/gatsby/dn/src/images/gatsby-astronaut.png


(sharp:61364): GLib-CRITICAL **: 07:29:30.084: g_hash_table_lookup: assertion 'hash_table != NULL' failed
```

**Comment by dante-blitz** [comment]
I figured out what was causing my issue. I am running Linux Subsystem on Windows 10. 

I was unaware that I needed to install Build Tools in order for sharp to work. Details are given here: https://blog.scottspence.me/wsl-setup

I hope this helps anyone else facing the same issue.

**Comment by Blackglade** [comment]
this issue is still relevant, I'm still experiencing this

**Comment by i-bsd** [comment]
Agree. Still an issue on FreeBSD.
The only way Gatsby is usable is to use drop `gatsby-plugin-sharp` back to an old version (2.0.7). This is not ideal or acceptable.

**Comment by PatrickResponsive** [comment]
Also still facing this issue

**Comment by dustinhorton** [comment]
I've run into this issue, but only in CI—so assumed it was a Linux issue since I've never seen it locally on macOS.

After some back and forth with the developer of sharp, he also recommended setting `sharp.cache(false)` in `gatsby-node.js`, along with `sharp.simd(true)` as previously discussed.

This _seems_ to have resolved the issue, but we're still toying with some CI settings.

**Comment by jasonbarry** [comment]
@dustinhorton could you post the relevant snippet of your `gatsby-node.js` file?

**Comment by KyleAMathews** [comment]
Is this a gatsby issue or a sharp issue? I believe we changed versions of sharp at some point?

**Comment by i-bsd** [comment]
> Is this a gatsby issue or a sharp issue? I believe we changed versions of sharp at some point?

I think it must be Gatsby-specific as it does not happen with Gridsome (Vue) which I believe, uses sharp.

> I've run into this issue, but only in CI—so assumed it was a Linux issue since I've never seen it locally on macOS.
> 
> After some back and forth with the developer of sharp, he also recommended setting `sharp.cache(false)` in `gatsby-node.js`, along with `sharp.simd(true)` as previously discussed.
> 
> This _seems_ to have resolved the issue, but we're still toying with some CI settings.

@dustinhorton I would also like to see your `gatsby-node.js` is possible. Thanks.

**Comment by KyleAMathews** [comment]
What version of sharp does gridsome use?

**Comment by dustinhorton** [comment]
@jasonbarry @i-bsd 

```javascript
const sharp = require('sharp')

sharp.cache(false)
sharp.simd(false)
```

**Comment by alexanderwe** [comment]
Maybe this is also related to https://github.com/gatsbyjs/gatsby/issues/6654 because the root of it seems to be https://github.com/gatsbyjs/gatsby/issues/14819 ?

**Comment by ericstumper** [comment]
For me the issue persisted on CI with an Alpine Docker Container even when I added the above mentioned sharp settings to `gatsby-node.js`. Container is running fine locally. I then added libvips as a global dependency to my Container which forces sharp to be compiled from source: Problem disappears.
Don't ask me why the precompiled binary fails only on CI inside the same Docker container...
For anyone using alpine linux: https://sharp.pixelplumbing.com/en/stable/install/

**Comment by sevonefive** [comment]
> I figured out what was causing my issue. I am running Linux Subsystem on Windows 10.
> 
> I was unaware that I needed to install Build Tools in order for sharp to work. Details are given here: https://blog.scottspence.me/wsl-setup
> 
> I hope this helps anyone else facing the same issue.

This worked for me also, just make sure you delete .cache and node_modules after install and re-run npm install.

**Comment by SatyaSudheer** [comment]
Updated .gitignore file with : 

# gatsby files
.cache/
public

it worked.

**Comment by SecondeJK** [comment]
Linux Mint 18 on an older HP Pavillion machine won't compile out of the box when running gatsby develop. Deleted the astronaut stock image, compiles.

**Comment by i-bsd** [comment]
> Linux Mint 18 on an older HP Pavillion machine won't compile out of the box when running gatsby develop. Deleted the astronaut stock image, compiles.

Drop `gatsby-plugin-sharp` back to version 2.0.7 and see if it compiles.

**Comment by SecondeJK** [comment]
@i-bsd Magic - rolling back sharp to 2.0.7 results in completed compile. Sounds like bug has been introduced then?

**Comment by i-bsd** [comment]
> @i-bsd Magic - rolling back sharp to 2.0.7 results in completed compile. Sounds like bug has been introduced then?

Seems to be. I still have not received an answer on this. Been asking about it for months now. I thought it was a FreeBSD-specific problem but since you're using Linux Mint, now I'm convinced it's not OS-related.

Would love to hear from devs on this if there's a way to fix it without rolling back `gatsby-plugin-sharp`.

**Comment by freemh** [comment]
Same problem here, I get an error message, even if I drop `gatsby-plugin-shar` back to version `2.7.0` 

```
Error: Failed to process image /home/mehdi/projects/gatsby-wordpress-acf-wpml/g
  atsby/gatsby-wordpress/src/images/gatsby-astronaut.png
```

I run Gatsby on a Ubuntu Server, and this is the `package.json`:

```
{
  "name": "gatsby-starter-default",
  "private": true,
  "description": "A simple starter to get up and developing quickly with Gatsby",
  "version": "0.1.0",
  "author": "Kyle Mathews <mathews.kyle@gmail.com>",
  "dependencies": {
    "gatsby": "^2.10.0",
    "gatsby-image": "^2.2.1",
    "gatsby-plugin-manifest": "^2.2.0",
    "gatsby-plugin-offline": "^2.2.0",
    "gatsby-plugin-react-helmet": "^3.1.0",
    "gatsby-plugin-sharp": "^2.2.1",
    "gatsby-source-filesystem": "^2.1.0",
    "gatsby-transformer-sharp": "^2.2.0",
    "prop-types": "^15.7.2",
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "react-helmet": "^5.2.1"
  },
  "devDependencies": {
    "prettier": "^1.18.2"
  },
  "keywords": [
    "gatsby"
  ],
  "license": "MIT",
  "scripts": {
    "build": "gatsby build",
    "develop": "gatsby develop",
    "format": "prettier --write src/**/*.{js,jsx}",
    "start": "npm run develop",
    "serve": "gatsby serve",
    "test": "echo \"Write tests! -> https://gatsby.dev/unit-testing\""
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/gatsbyjs/gatsby-starter-default"
  },
  "bugs": {
    "url": "https://github.com/gatsbyjs/gatsby/issues"
  }
}
```

**Comment by freemh** [comment]
I solve the problem by following these steps:
```
rm -rf node_modules
rm package-lock.json yarn.lock
npm cache clear --force
```
Then I install `libpng-dev`:

```
sudo apt install -y build-essential libpng-dev
```
And at the end, I run an `npm install` and a `gatsby develop`.

I think that the problem is related to the operating system not Gatsby, I'm running gatsby from a Ubuntu Server installed as a virtual machine:
```
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.2 LTS
Release:	18.04
Codename:	bionic
```

**Comment by HrishiPolAleph** [comment]
@freemh  by removing the files and npm install solved the problem for me on macos.

**Comment by harrygreen** [comment]
Apologies if this is unrelated. We're hitting a similar issue when running builds after upgrading to Gatsby 2.9 and above. I've tried downgrading `gatsby-plugin-sharp` to 2.0.7 as mentioned above - no help there. 

```
success Building static HTML for pages - 9.744 s — 8/8 13.24 pages/second
[=========================== ]   81.767 s 58/60 97% Generating image thumbnails
(sharp:10197): GLib-CRITICAL **: 10:39:11.192: g_hash_table_lookup: assertion 'hash_table != NULL' failed
```

Interestingly, I've had some success by injecting a small delay into `gatsby-plugin-sharp`'s [`runJob`](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-sharp/src/scheduler.js#L153) queue.

**Comment by sheelah** [comment]
I'm seeing the same error during builds as @harrygreen, with `gatsby` v2.9.11 and `gatsby-plugin-sharp` v2.1.9.  This is cropping up on MacOS as well as Netlify builds.

**Comment by harrygreen** [comment]
@sheelah did you find a workaround / fix?

**Comment by sheelah** [comment]
> @sheelah did you find a workaround / fix?

Not yet. Did you? Did notice that after updating all project dependencies (& node version since why not) it's no longer core dumping after that error as it was before.  Still seems not ideal though.

**Comment by harrygreen** [comment]
@sheelah unfortunately no luck yet. Will take another look soon. My hack was on this line: https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-sharp/src/scheduler.js#L153. I changed
```js
Promise.all(promises).then(() => {
```
to
```js
Promise.all([...promises, new Promise(resolve => setTimeout(resolve, 10000))]).then(() => {
```
I have no idea why it worked. So I don't recommend it!

**Comment by sheelah** [comment]
@harrygreen Ah - gotcha. Well thanks for the snippet! Will keep an eye out for other ideas.

**Comment by birjolaxew** [comment]
Relevant discussion in the sharp repo: https://github.com/lovell/sharp/issues/409

It looks like it happens when the process is killed before sharp has finished resizing. That would also explain why adding a timeout would fix it (by allowing the resizing to finish).

**Comment by smurrayatwork** [comment]
We were having this same issue and discovered it was being caused due to a few images being corrupt. When gatsby-plugin-sharp encounters a corrupted image, the errors related to it aren't caught for some reason. As soon as we fixed those images, this issue was resolved.

If you're processing jpg images, [try a tool like jpeginfo](https://www.commandlinefu.com/commands/view/2352/find-corrupted-jpeg-image-files) to see if your images are okay.

It's available on MacOS via brew.

**Comment by derakhshanfar** [comment]
same issue for me, did you find a workaround / fix?

**Comment by sheelah** [comment]
I never found a permanent fix for this.  Did run `jpeginfo` and I didn't find any corrupted images.

**Comment by ciruz** [comment]
@sheelah same for us, we had exactly the same error and jpeginfo found no corrupted images. This error was the reason we switched to next.js

**Comment by smurrayatwork** [comment]
Sorry - I should have mentioned that you could also have corrupt images of other types, such as pngs, which that tool wouldn't cover. For PNGs I used the `pngfix` program to test them, which you can get from libpng tools.

**Comment by tw1t611** [comment]
got the same error on debian, while only using .jpg files.

**Comment by bogdancss** [comment]
We are running the latest packages as of today (gatsby `2.18.0`, gatsby-image `2.2.33`, gatsby-plugin-sharp `2.3.2` & gatsby-transformer-sharp `2.3.5`) and trying to build our project locally (MacOS) and are encountering the same issues here.

It fails randomly while generating image thumbnails (roughly 80k thumbs):

` [=--------] 5568/84161 480.7 s 6% Generating image thumbnails`
`zsh: segmentation fault  gatsby build`

**Comment by dburrows** [comment]
Had this today on a pristine gatsby site generated via:

```
gatsby new my-themed-blog https://github.com/gatsbyjs/gatsby-starter-blog-theme
``

Not a great newbie experience :(
OS X 10.14.6, Node v11.14.0, MacBookPro

**Comment by smurrayatwork** [comment]
I've noticed that this issue can happen when any type of file extension is corrupted. The tool I had linked to was only for jpg, so you might be having this issue with png files, or any of the supported types sharp processes.

**Comment by dburrows** [comment]
Thanks @smurrayatwork but I think I've narrowed it down to a clash with the Anaconda Python distro, seems to not play well with `node-gyp` and `sharp` - removing the Anaconda setup from my `.bash_profile` (in mine it's outlined by 'conda init` comments)  fixed all the issues.

**Comment by Vacilando** [comment]
Experiencing this in develop mode under WSL since a few days. Not on Netlify. Latest Gatsby and NPM.
`gatsby clean` not helpful.

**Comment by mad-gav** [comment]
We are facing this issue on a repo. Seems to build fine on netlify but can't dev or build locally.

Have tried gatsby clean, fresh install. Upgrading gatsby, downgrading sharp. Adding the sharp into the gatsby node

**Comment by birjolaxew** [comment]
@mad-gav Looks like libvips is complaining about a corrupt JPEG file. Can you provide the image in question?

**Comment by mad-gav** [comment]
I've tried updating image sharp. It breaks on different images. Whats strange is it builds every single time on netlify?!?

**Comment by mad-gav** [comment]
All the images are coming from prismic. E.g this is one it broke on

https://images.prismic.io/greatstate/a02c023b-f454-4a57-9748-6bd5a0cacee3_Partners-9.jpg?auto=compress,format
![Screenshot 2020-02-20 at 14 15 40](https://user-images.githubusercontent.com/60468535/74941762-801c9c00-53eb-11ea-8fd1-9701539a759d.png)

**Comment by mad-gav** [comment]
I still don't understand why this same repo will build every single time on netlify but locally i get the error above?

**Comment by birjolaxew** [comment]
@mad-gav Nor does anyone else here. I have a couple wild guesses:

1. Could be that there is a mess up caused by Window's case-insensitive filesystem. Are you sure the name on disk is exactly the same as the name Netlify reports in its logs, capitalization included?

2. Could be that your images are corrupt. Have you checked with jpeginfo?

3. Could be as simple as being the same issue described in the OP here. Have you tried increasing the timeout?

In the end you're the only one who can debug this, since you're the only one who has access to your failing test case.

**Comment by Vacilando** [comment]
In my case, which I never solved the answers are:

1. No. It works with all the images but only in smaller batches.
2. Ditto.
3. Increase timeout? How - where is it documented?

**Comment by birjolaxew** [comment]
@Vacilando That suggestion comes from [this comment](https://github.com/gatsbyjs/gatsby/issues/13442#issuecomment-510806796). If you're running this on WSL, I'm also assuming that you've already followed the other comments in this issue and [installed `build-essentials`](https://github.com/gatsbyjs/gatsby/issues/13442#issuecomment-505220302).

**Comment by mad-gav** [comment]
I've gone in an increased the timeout and now get lots of timeouts

![image](https://user-images.githubusercontent.com/60468535/75142926-fff87e00-56eb-11ea-8513-882d48ac7f60.png)

Do you think it could be to do with prismic providing all images with query strings and commas at the end?

**Comment by mad-gav** [comment]
Looks like netlify is running it on a linux machine. We have all devs here on MacOS and we can't get it to build

**Comment by mad-gav** [comment]
I've tried a lot of things now i think we're going to have to rip out all the image processing that's done by sharp. I cannot get it to work locally

**Comment by elias-thok** [comment]
Has anyone ever figured out how to fix this for good?

I feel like all I do when I try to build something with Gatsby is fixing these weird errors, deleting node_modules, caches and all sorts of random stuff to get it to work. Very frustrating tbh.

I did the following and it works for now:

* globally & locally updated all npm packages with `npm update`
* clear global NPM cache with `npm cache clear --force`
* Delete node_modules, package-lock.json & .cache in project folder
* run a fresh `npm install`

```
System:
    OS: macOS 10.15.3
  Binaries:
    Node: 13.11.0 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.3 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
```

**Comment by TravisHi** [comment]
@elias-thok following your instructions worked for me. It seems like my issue related to changing local environment.

**Comment by mad-gav** [comment]
The build appears to work now. Nothing has changed content or code wise. Wondering if either prismic have improved performance or the connection is faster so not getting any timesouts on the images? Very strange

**Comment by RavenColEvol** [comment]
I've tried converting CRA to gatsby . I'm now trying gatsby-image but I'm getting this error while querying image from graphql. 
I've already installed gatsby filesystem and configured it.

`
plugins: [
    `gatsby-plugin-react-helmet`,
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: "images",
        path: `${__dirname}/src/static`
      }
    },
    `gatsby-transformer-sharp`,
    `gatsby-plugin-sharp`,
    {
      resolve: `gatsby-plugin-styled-components`,
      options: {
        displayName: false
      }
    },
    
  ]
`


Error which I'm getting when i query any image.

`
 ERROR

Processing /mnt/c/Users/oneco/Desktop/Ravi/Coding/React/react-portfolio/src/static/js.png failed

Original error:




  WorkerError: Processing /mnt/c/Users/oneco/Desktop/Ravi/Coding/React/react-portfolio/src/static/js.png failed
  Original error:

  - jobs-manager.js:314 exports.enqueueJob
    [react-portfolio]/[gatsby]/dist/utils/jobs-manager.js:314:23

  - runMicrotasks

  - task_queues.js:97 processTicksAndRejections
    internal/process/task_queues.js:97:5


not finished Generating image thumbnails - 0.695s


———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
4 pages                                                                                            In Progress                                                                                            portfolio
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! portfolio@0.1.0 develop: `gatsby develop`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the portfolio@0.1.0 develop script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ravi/.npm/_logs/2020-03-28T07_20_39_537Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! portfolio@0.1.0 start: `npm run develop`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the portfolio@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ravi/.npm/_logs/2020-03-28T07_20_39_575Z-debug.log
`

**Comment by pvdz** [comment]
Dear everyone. The original problem in this issue was caused by Sharp emitting warnings or even segfaulting. This has since been fixed in Sharp 0.25 and Gatsby has been shipping this for over a month now. If you're still seeing warnings or segfaults relating to large sets of images, make sure you use an updated version of sharp related gatsby packages.

If you see this issue again please don't hesitate to post a new issue about it (and feel free to explicitly tag me).

It seems the WSL issue is something different, unrelated to the original problem. If that's an actual issue please open a new issue for it.

I'm going to close this ticket and lock it to prevent more users tacking unrelated reports to it. Thank you!

**Comment by jamessimone** [comment]
Something to note - for posterity, since this is the only page that turns up when searching for the `Error: VipsJpeg; Premature end of JPEG file` error in regards to Gatsby: if you are still running into this issue after upgrading to the newer Gatsby packages, you probably actually _do_ have a corrupt image file. 

I just encountered this issue on a client's site (which had built, successfully, several times since the corrupt image was put into place, including with the Sharp upgrades - hence my confusion). Try using [jpeginfo](https://github.com/tjko/jpeginfo), or [BadPeggy](https://www.coderslagoon.com/) on Windows, to scan the images on your site if that's the case. On a large site with plenty of images, it would be difficult to diagnose the issue directly since the image causing the error is not always printed out to the console.

Hopefully that helps if you wound up here as a result of trying to investigate the premature end of JPEG file error message.
